lib/soap.js

```diff
@@ -440,18 +444,18 @@ module.exports = class Soap {
         try {
             // need to wait as it may error
             return await _parseResponse(response, options.key);
-        } catch (ex) {
-            if (ex.message === 'Token Expired' && remainingAttempts) {
+        } catch (error) {
+            if (error.message === 'Token Expired' && remainingAttempts) {
                 // force refresh due to url related issue
                 await this.auth.getAccessToken(true);
                 // set to no more retries as after token refresh it should always work
                 return this._apiRequest(options, 1);
-            } else if (ex instanceof SOAPError) {
+            } else if (error instanceof SOAPError) {
                 //rethrow as is already handled/parsed
-                throw ex;
+                throw error;
             } else {
                 //unknown error
-                throw new SOAPError(ex, response, null);
+                throw new SOAPError(error, response, undefined);
```

Comment:
checked what SOAPError is expecting and testing for in it's 3rd parameter?

Reply:
it is a safe change for the error, realistically, i would be happy to take risk here on error handling, but if you are concerned, can also move to v1 change

Reply:
bad wording on my part: "did YOU check..."? :D if things worked fine in your tests then I'm good. Though I gotta admit I don't understand why null has to become undefined. We've been using null across projects for a while now in these cases

Reply:
did you? @DougMidgley

Reply:
Yes tested while running the rate limiting tests without issue.